pkg/files/file.go

```diff
@@ -225,14 +228,32 @@ func (r *File) IsTemplate() bool {
 	if r.markedTemplate != nil {
 		return *r.markedTemplate
 	}
-	return r.isTemplate()
+	return r.shouldProcess() && r.hasTemplating()
 }
 
-func (r *File) isTemplate() bool {
+func (r *File) shouldProcess() bool {
 	t := r.Type()
 	return !r.IsLibrary() && (t == TypeYAML || t == TypeText)
 }
 
+func (r *File) hasTemplating() bool {
+	data, err := r.Bytes()
+	if err != nil {
+		panic(fmt.Sprintf("Unable to get contents of file %s: %s", r.OriginalRelativePath(), err))
+	}
+	switch r.Type() {
+	case TypeYAML:
+		for _, ann := range literal.AllAnnotations {
+			if bytes.Contains(data, ann) {
+				return true
+			}
+		}
+	case TypeText:
+		return bytes.Contains(data, literal.TextTemplateStartToken)
```

Comment:
hmm... i am not sure we can do this for text (== always template by default) since there is no such thing as a comment in text.

in the line above, what if user only placed `@)` and screwed up opening tag?

Reply:
... and there's no need to change how text files are processed. This wasn't asked for.